pkg/job/scrape_test.go

```diff
@@ -0,0 +1,87 @@
+package job
+
+import (
+	"testing"
+
+	"github.com/brianvoe/gofakeit/v6"
+
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/model"
+)
+
+func BenchmarkSlices(b *testing.B) {
+	type testcase struct {
+		jobs          int
+		resultsPerJob int
+	}
+
+	for name, tc := range map[string]testcase{
+		"1 job large results": {
+			jobs:          1,
+			resultsPerJob: 500000,
+		},
+		"1 job medium results": {
+			jobs:          1,
+			resultsPerJob: 100000,
+		},
+		"1 job small results": {
+			jobs:          1,
+			resultsPerJob: 1000,
+		},
+		"multiple jobs large results": {
+			jobs:          5,
+			resultsPerJob: 100000,
+		},
+		"multiple jobs medium results": {
+			jobs:          5,
+			resultsPerJob: 10000,
+		},
+		"multiple jobs small results": {
+			jobs:          5,
+			resultsPerJob: 100,
+		},
+	} {
+		input := setupInputData(tc.jobs, tc.resultsPerJob)
+
+		b.Run(name+" slice of slices", func(b *testing.B) {
+			doSliceOfSlicesBench(b, input)
+		})
+
+		b.Run(name+" slice concat", func(b *testing.B) {
+			doSliceConcatBench(b, input)
+		})
+	}
+}
```

Comment:
I don't think this needs to stay just wanted to show the code behind the benchmarks in case I did something that would invalidate the results.